analyzers/src/SonarAnalyzer.CSharp/Rules/UseAwaitableMethod.cs

```diff
@@ -158,7 +159,8 @@ private static WellKnownExtensionMethodContainer BuildWellKnownExtensionMethodCo
             }
             if (original == awaitableRoot && result is ExpressionSyntax resultExpression)
```

Comment:
Just a suggestion no need to apply if you disagree: I cannot leave a comment in previous lines but it might be better to extract the function given as input to `ReplaceNodes` as a local method - for readability.

Reply:
Good idea: dda3dd8a1c9a56654b4fedb5d474188ab102f32e